python/rmm/_cuda/gpu.py

```diff
@@ -1,59 +1,36 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
 
-from rmm._cuda.gpu cimport (
-    CUresult,
-    cudaDeviceAttr,
-    cudaDeviceGetAttribute,
-    cudaDeviceProp,
-    cudaDriverGetVersion,
-    cudaError,
-    cudaError_t,
-    cudaGetDeviceCount,
-    cudaGetDeviceProperties,
-    cudaGetErrorName,
-    cudaGetErrorString,
-    cudaRuntimeGetVersion,
-    cuDeviceGetName,
-    cuGetErrorName,
-    cuGetErrorString,
-)
-
-from enum import IntEnum
+from cuda import cuda, cudart
 
 
 class CUDARuntimeError(RuntimeError):
-
-    def __init__(self, cudaError_t status):
+    def __init__(self, status: cuda.CUresult.CUDA_SUCCESS):
         self.status = status
-        cdef str name = cudaGetErrorName(status).decode()
-        cdef str msg = cudaGetErrorString(status).decode()
+        _, name = cudart.cudaGetErrorName(status)
+        _, msg = cudart.cudaGetErrorString(status)
         super(CUDARuntimeError, self).__init__(
-            '%s: %s' % (name, msg))
+            "%s: %s" % (name.decode(), msg.decode())
+        )
 
     def __reduce__(self):
         return (type(self), (self.status,))
 
 
 class CUDADriverError(RuntimeError):
-
-    def __init__(self, CUresult status):
+    def __init__(self, status: cuda.CUresult):
```

Comment:
```suggestion
    def __init__(self, status: cuda.CUresult):
```

This API leaks  a CUDA Python  dependency. It would be nice is `status` was a type that didn't depend on  CUDA Python. Investigating.

Reply:
Can you explain this problem a little further? Is this something specific to typing support?

Reply:
This module provides higher level abstractions around CUDA Python that downstream libs like cuDF, cuGraph are using. I found that replacing those high level API calls downstream with low-level CUDA Python calls is unweidly. So I'm opting to keep this module.

However, this particular class is a leaky abstraction, and pushes the CUDA Python dependency downstream. I'd like to avoid that.

Does that help clarify, and do you think this worth doing?

Reply:
Maybe I'm missing something obvious here -- where is the leak?

Since cuda-python will be a dependency of rmm, I don't see how this would be a new problem compared to the previous implementation of the `CUDADriverError` class.

Reply:
This is essentially asking whether all public RMM APIs that effectively use lower-level CUDA objects need to wrap them to avoid users having to interact with the `cuda` package directly, right? We already wrap concepts like streams in RMM. Are errors and statuses the only exceptions at the moment? Do we foresee any others in the future?

IMO while it would be easy to wrap error/status types into RMM-specific objects, I don't think we want to paint ourselves into a corner by claiming that we will always avoid using CUDA Python objects in RMM's API unless we're _very_ sure that this won't introduce onerous wrapping requirements in the future.

Reply:
Following up - @shwina and I discussed and came to a similar conclusion.

It's fine for rmm to "leak" the cuda-python API here because we hope that eventually this part of rmm can be replaced entirely by a higher-level object-oriented API in cuda-python. At that point, there will be a (direct/non-transitive) dependency on cuda-python in the downstream packages currently using this rmm API, and it would hopefully be a matter of just changing the functions from rmm's high-level APIs to cuda-python's high-level APIs.

"Leaking API" in a way that requires using cuda-python APIs as arguments or return types for some of these rmm APIs has two effects. First, it introduces that direct dependency on cuda-python into downstream packages a bit sooner (not a bad thing, since it will already be a transitive dependency). Secondly, and more importantly, it allows this layer of rmm to avoid re-wrapping cuda-python's enums and other APIs. That minimizes the cost/burden placed on rmm to provide a bridge until cuda-python provides higher level APIs.